File: pdfsecondmonitor/ctlMovie.vb
Public Class ctlMovie
    Private _dispacher As FormDispacher = FormDispacher.GetInstance


    Private player As AxWMPLib.AxWindowsMediaPlayer

    Private _fileViewParam As FileViewParam

    Public Sub SetFileInfo(f As FileViewParam)
        Me._fileViewParam = f
        If _fileViewParam Is Nothing Then
            Return
        End If
        player = _dispacher.ShowMovie()

        player.URL = _fileViewParam.FileName
        player.uiMode = "none"
        player.stretchToFit = True

    End Sub

#Region "Media Playerの処理"

    Private inPlay As Boolean = False


    Private Sub btnStartStop_Click(sender As Object, e As EventArgs) Handles btnStartStop.Click
        player.Ctlcontrols.play()
    End Sub

    Private Sub btnStop_Click(sender As Object, e As EventArgs) Handles btnStop.Click
        player.Ctlcontrols.pause()
    End Sub

    Private Sub Button5_Click(sender As Object, e As EventArgs) Handles btnFastForward.Click
        player.Ctlcontrols.fastForward()
    End Sub

    Private Sub btnFastReverse_Click(sender As Object, e As EventArgs) Handles btnFastReverse.Click
        player.Ctlcontrols.fastReverse()
    End Sub


    Public Sub ControlEnabled()

        Me.Enabled = Not (_fileViewParam Is Nothing)

    End Sub






#End Region
End Class

File: MediaToSecondMonitor/ctlMovie.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Drawing;
using System.Data;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace MediaToSecondMonitor
{
    public partial class ctlMovie : UserControl
    {
        public ctlMovie()
        {
            InitializeComponent();
        }
        private FormDispacher _dispacher = FormDispacher.Instance;


        private AxWMPLib.AxWindowsMediaPlayer player;

        private FileViewParam _fileViewParam;

        public void SetFileInfo(FileViewParam f)
        {
            this._fileViewParam = f;
            if (_fileViewParam == null)
                return;
            player = _dispacher.GetPlayer();

            player.URL = _fileViewParam.FileName;
            player.uiMode = "none";
            player.stretchToFit = true;
        }


        private bool inPlay = false;





        public void ControlEnabled()
        {
            this.Enabled = !(_fileViewParam == null);
        }

        private void BtnPlay_Click(object sender, EventArgs e)
        {
            if(inPlay)
            {
                player.Ctlcontrols.pause(); 
            }
            else
            {
                player.Ctlcontrols.play();
            }
            inPlay = !inPlay;

        }



        private void btnFastReverse_Click(object sender, EventArgs e)
        {
            player.Ctlcontrols.fastReverse();
        }

        private void btnForward_Click(object sender, EventArgs e)
        {

            player.Ctlcontrols.fastForward();
        }

        private void btn_Click(object sender, EventArgs e)
        {
            player.Ctlcontrols.stop();
            player.Ctlcontrols.play();
            player.Ctlcontrols.pause();
            inPlay = false;
        }
    }
}

